lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_returning_widgets/visitor.dart

```diff
@@ -4,29 +4,51 @@ class _Visitor extends RecursiveAstVisitor<void> {
   final _declarations = <Declaration>[];
 
   final Iterable<String> _ignoredNames;
+  final Iterable<String> _ignoredAnnotations;
 
   Iterable<Declaration> get declarations => _declarations;
 
-  _Visitor(this._ignoredNames);
+  _Visitor(this._ignoredNames, this._ignoredAnnotations);
 
   @override
   void visitMethodDeclaration(MethodDeclaration node) {
     super.visitMethodDeclaration(node);
 
-    if (!node.isSetter && !_isIgnored(node.name.name)) {
```

Comment:
I found two duplicate code blocks in `visitMethodDeclaration` and `visitFunctionDeclaration` so decided to move common implementation into `_visitDeclaration` method.

Reply:
Yeah, looks good!